captum/attr/_core/deep_lift.py

```diff
@@ -27,11 +27,30 @@
 
 # Check if module backward hook can safely be used for the module that produced
 # this inputs / outputs mapping
-def _check_valid_module(inputs, outputs):
+def _check_valid_module(inputs_grad_fns, outputs):
+    def is_output_cloned(output_fn, input_grad_fn):
```

Comment:
For the layer case as the output was cloned it didn't match with the gradient_fn from input. I had to look into next functions to find the origial of the clone. 
cc: @vivekmig